virttest/qemu_vm.py

```diff
@@ -2305,6 +2305,9 @@ def sort_key(dev):
             options["deflate-on-oom"] = deflate_on_oom
             guest_polling = balloon_params.get("balloon_opt_guest_polling")
             options["guest-stats-polling-interval"] = guest_polling
+            balloon_addr = balloon_params.get("balloon_addr")
+            if balloon_addr:
+                options["addr"] = balloon_addr
```

Comment:
Looks it only works for PCI buses, however, wo should also consider about other bus types (for aarch64, s390x..)

Reply:
@luckyh I guess other arch wont need this option being specified, address pattern should be always automatically generated, only for migration + hotplug it needs to be specified by user. Or better we define only x86 and ppc for such balloon case.

Reply:
In case of wrong usage, maybe filter the bus type like`if isinstance(balloon_bus, qdevices.QPCIBus) and balloon_addr:`, what do you think? @luckyh

Reply:
I am still not clear about the primary use case for introducing this feature. If it is used to workaround the address mismatch issue of migration, then I would give **NACK** for this, because obviously, we have a better solution: let VT allocate the address automatically and mark it up in qemu command line.

If it is supposed to let user be able to customize balloons' bus address, I would accept, but I do request it works for all the supported platforms.

Thanks.

Reply:
Btw, it's the time to stress the necessity of writing a reasonable commit message.

Reply:
Actually,  on ppc platform, **hotplug nic device + migration** also hit this issue. It is necessary to specify pci bus and address when we hot-plug a device, and it needs to match the divece bus info of the dst guest if we want to migrate it.
Agree with @luckyh that let VT allocate the address automatically

Reply:
As to the supporting for other platforms,  I need do some research first, thanks for your reminder.

Reply:
@PaulYuuu That does match the test scenario. We have hotplug before migration scenario for almost all device test.

Reply:
@huangyum so it turns out that there are bugs in `vm.clone()`: it does not reserve the addresses correctly for initial devices, as @PaulYuuu  mentioned, the issue is not only against balloon tests, but also nics, blocks, .. so for this kind of issue we need a common solution.

Reply:
@luckyh In balloon cases, it does not only not reserve the address but also the balloon device itself. I see some workaround lines in balloon_hotplug.py.
'''
 84         # temporary workaround for migration
 85         vm.params["balloon"] = "balloon%d" % idx
 86         vm.params["balloon_dev_devid"] = "balloon%d" % idx
 87         vm.params["balloon_dev_add_bus"] = "yes"
 88         devs = vm.devices.get_by_params({"id": 'balloon%d' % idx})
 89         vm.params["balloon_pci_bus"] = devs[0]["bus"]
'''
Without these lines, dst qemu won't boot with a balloon device at all.  I'm not sure about other devices.
Besides, whether we have a common solution or not, this patch should be a must to support specifying address for balloon device. Thanks.